### PR TITLE
Fixes #26777 - added taxonomy_taxable id and type index

### PR DESCRIPTION
**Pull request by lzap**
Reported by a customer.

### CODE REVIEW

**Review by timogoebel — DISMISSED** [review]
Makes sense. Applied this to our prod environment. Let's see how this performs.

**Comment by timogoebel** [comment]
@tbrisker: Mind taking a look if you have a second?
I added this in our prod instance. Works fine, I don't have any data to share unfortunately.

File: db/migrate/20190510082811_add_taxonomy_id_taxable_type_indices_to_taxable_taxonomies.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class AddTaxonomyIdTaxableTypeIndicesToTaxableTaxonomies < ActiveRecord::Migration[5.2]
+  def change
+    change_table :taxable_taxonomies do |t|
+      t.index ["taxonomy_id", "taxable_type"], name: "index_taxable_taxonomy_id_type"
```
Comment (tbrisker) [inline db/migrate/20190510082811_add_taxonomy_id_taxable_type_indices_to_taxable_taxonomies.rb]: Is the name needed? rails auto generates one that it knows to remove later.

**Review by tbrisker — COMMENTED** [review]
if we add this index, i think we can drop this one as dbs know to use partial prefix in indexes:
```   t.index ["taxonomy_id"], name: "index_taxable_taxonomies_on_taxonomy_id"```

File: db/migrate/20190510082811_add_taxonomy_id_taxable_type_indices_to_taxable_taxonomies.rb (hunk 1 above)
Comment (lzap) [reply db/migrate/20190510082811_add_taxonomy_id_taxable_type_indices_to_taxable_taxonomies.rb]: Yeah I've seen it in previous migrations so I thought it's our convention. Let's drop it then.

**Comment by lzap** [comment]
Rebased, table after the new migration:

```ruby
  create_table "taxable_taxonomies", force: :cascade do |t|
    t.integer "taxonomy_id"
    t.integer "taxable_id"
    t.string "taxable_type", limit: 255
    t.datetime "created_at"
    t.datetime "updated_at"
    t.index ["taxable_id"], name: "index_taxable_taxonomies_on_taxable_id"
    t.index ["taxable_type", "taxable_id", "taxonomy_id"], name: "taxable_index", unique: true
    t.index ["taxonomy_id", "taxable_type"], name: "index_taxable_taxonomies_on_taxonomy_id_and_taxable_type"
  end
```

Table after rollback:

```ruby
  create_table "taxable_taxonomies", force: :cascade do |t|
    t.integer "taxonomy_id"
    t.integer "taxable_id"
    t.string "taxable_type", limit: 255
    t.datetime "created_at" 
    t.datetime "updated_at"
    t.index ["taxable_id"], name: "index_taxable_taxonomies_on_taxable_id"
    t.index ["taxable_type", "taxable_id", "taxonomy_id"], name: "taxable_index", unique: true
    t.index ["taxonomy_id"], name: "index_taxable_taxonomies_on_taxonomy_id"
  end
```

File: db/migrate/20190510082811_add_taxonomy_id_taxable_type_indices_to_taxable_taxonomies.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class AddTaxonomyIdTaxableTypeIndicesToTaxableTaxonomies < ActiveRecord::Migration[5.2]
+  def up
+    remove_index :taxable_taxonomies, name: "index_taxable_taxonomies_on_taxonomy_id"
```
Comment (tbrisker) [inline db/migrate/20190510082811_add_taxonomy_id_taxable_type_indices_to_taxable_taxonomies.rb]:
```suggestion
    remove_index :taxable_taxonomies, :taxonomy_id
```

File: db/migrate/20190510082811_add_taxonomy_id_taxable_type_indices_to_taxable_taxonomies.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    add_index :taxable_taxonomies, ["taxonomy_id", "taxable_type"]
+  end
+
+  def down
+    remove_index :taxable_taxonomies, ["taxonomy_id", "taxable_type"]
+    add_index :taxable_taxonomies, ["taxonomy_id"], name: "index_taxable_taxonomies_on_taxonomy_id"
```
Comment (tbrisker) [inline db/migrate/20190510082811_add_taxonomy_id_taxable_type_indices_to_taxable_taxonomies.rb]:
```suggestion
    add_index :taxable_taxonomies, :taxonomy_id
```

**Review by tbrisker — CHANGES_REQUESTED** [review]
no need to use the name on the existing index either - it is also an autogenerated name

**Comment by lzap** [comment]
Ok squashed, this won't work :-)

**Comment by ares** [comment]
what's the status here? [test foreman] [test katello] [test upgrade]

**Review by tbrisker — APPROVED** [review]
Thanks @lzap !

**Review by tbrisker — CHANGES_REQUESTED** [review]
looks like mysql failures are related - it can't drop the index because it's used a foreign key

**Comment by lzap** [comment]
What now? I suggest simply to add the index and leave the single-column there.

How is possible that we've dropped many of these during your cleanup @tbrisker and MySQL was not complaining?

**Comment by tbrisker** [comment]
@lzap i wonder if maybe reversing the order (i.e. first adding the new index and then removing the old) would fix that issue? if not I guess we'll need to live with both as long as we support mysql.

**Comment by lzap** [comment]
Trying the reverse order.

**Review by tbrisker — DISMISSED** [review]
Looks like that did it. Test failure unrelated, but I can't merge from my phone because of it.

**Review by mmoll — APPROVED** [review]

**Comment by mmoll** [comment]
merged, díky @lzap!

**Pull request merged**